src/segmentmytif/main.py

```diff
@@ -1,33 +1,88 @@
 import argparse
 from pathlib import Path
+from typing import Literal
 
 import numpy as np
 import dask.array as da
 from numpy import ndarray
 from sklearn.ensemble import RandomForestClassifier
+import geopandas as gpd
+import rioxarray
+import dask
+import xarray as xr
 
 from segmentmytif.features import get_features, FeatureType
 from segmentmytif.logging_config import setup_logger, log_duration, log_array
 from segmentmytif.utils.io import read_geotiff, save_tiff
+from segmentmytif.utils.geospatial import get_label_array
 
 logger = setup_logger(__name__)
 
 
-def read_input_and_labels_and_save_predictions(input_path: Path, labels_path: Path, output_path: Path,
-                                               feature_type=FeatureType.FLAIR, features_path: Path = None,
-                                               **extractor_kwargs) -> None:
+def read_input_and_labels_and_save_predictions(
+    raster_path: Path,
+    pos_labels_path: Path,
+    neg_labels_path: Path,
+    output_path: Path,
+    feature_type=FeatureType.IDENTITY,
+    features_path: Path = None,
+    mode: Literal["normal", "parallel", "safe"] = "normal",
+    chunks: dict = None,
+    chunk_overlap: int = 16,
+    **extractor_kwargs,
+) -> None:
     logger.info("read_input_and_labels_and_save_predictions called with the following arguments:")
     for k, v in locals().items():
         logger.info(f"{k}: {v}")
 
-    input_data, profile = read_geotiff(input_path)
-
-    features = get_features(input_data, input_path, feature_type, features_path, profile, **extractor_kwargs)
-
-    labels, _ = read_geotiff(labels_path)
-    prediction_map = make_predictions(features, labels)
-
-    save_tiff(prediction_map, output_path, profile)
+    if chunks is None:
+        chunks = {"band": 1, "x": 1000, "y": 1000}
```

Comment:
If the chunk size (including overlap I guess) is not divisible by 16, we'll be padding before doing feature extraction. This actually takes a little bit of compute. This will be logged to `INFO` to the terminal for instance. E.g.
```
2025-01-28 16:46:07,130 - segmentmytif.features - INFO - Added temporary padding for band 0: (original 1210 x 718) -> (padded 1216 x 720)
```
Maybe, we can choose the _default_ chunk size in such a way that it doesn't need padding. 1008 for instance, but 1024 may look a little nicer and might raise fewer questions ;-)

```suggestion
        chunks = {"band": 1, "x": 1024, "y": 1024}
```